sensor/common/metrics/metrics.go

```diff
@@ -260,6 +267,39 @@ var (
 		Name:      "num_messages_waiting_for_transmission_to_central",
 		Help:      "A counter that tracks the operations in the responses channel",
 	}, []string{"Operation", "MessageType"})
+
+	// componentProcessMessageDurationSeconds tracks the duration of ProcessMessage calls for each component
+	componentProcessMessageDurationSeconds = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "component_process_message_duration_seconds",
+		Help:      "Time taken to process messages from Central in each sensor component",
+		Buckets:   prometheus.ExponentialBuckets(0.001, 2, 12), // 1ms to ~4s
+	}, []string{ComponentName})
+
+	// ComponentQueueMessagesDroppedCount tracks the number of messages dropped due to size limit
+	ComponentQueueMessagesDroppedCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "component_queue_messages_dropped_count",
+		Help:      "Number of messages dropped due to full queue",
+	}, []string{ComponentName})
+
+	// ComponentQueueOperations keeps track of the operations of the component queue buffer.
+	ComponentQueueOperations = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "component_queue_operations_total",
+		Help:      "A counter that tracks the number of ADD and REMOVE operations on the component buffer queue. Current size of the queue can be calculated by subtracting the number of remove operations from the add operations",
+	}, []string{ComponentName, "Operation"})
```

Comment:
Why not changing the `ComponentQueueMessagesDroppedCount` into a `DROP` operation of this metric?